src/frontend/packages/suse-extensions/src/custom/kubernetes/services/kubernetes-endpoint.service.ts

```diff
@@ -65,6 +80,50 @@ export class KubernetesEndpointService {
     this.constructCoreObservables();
   }
 
+  getCaaspData(nodes$: Observable<KubernetesNode[]> = this.nodes$): Observable<CaaspData> {
+    return nodes$.pipe(
+      map(nodes => {
+        const info: CaaspData = {
+          version: 'Unknown',
+          versionMismatch: false,
+          updates: 0,
+          disruptiveUpdates: 0,
+          securityUpdates: 0
+        };
+        const versions = {};
+
+        nodes.forEach(n => {
+          if (n.metadata && n.metadata.annotations) {
+            if (n.metadata.annotations[CAASP_VERSION_ANNOTATION]) {
+              const version = n.metadata.annotations[CAASP_VERSION_ANNOTATION];
+              if (!versions[version]) {
+                versions[version] = 0;
+              }
+              versions[version]++;
+            }
+
+            info.updates += this.hasBooleanAnnoation(n.metadata.annotations, CAASP_HAS_UPDATES_ANNOTATION)
+            info.disruptiveUpdates += this.hasBooleanAnnoation(n.metadata.annotations, CAASP_DISRUPTIVE_UPDATES_ANNOTATION)
+            info.securityUpdates += this.hasBooleanAnnoation(n.metadata.annotations, CAASP_DISRUPTIVE_UPDATES_ANNOTATION)
+          }
+        });
+
+        if (Object.keys(versions).length === 0) {
+          return null;
+        }
+
+        info.version = Object.keys(versions).join(', ');
+        info.versionMismatch = Object.keys(versions).length !== 1;
+        return info;
+      })
+    )
+  }
+
+  // Check for the specified annotation with a value other than 'no'
+  private hasBooleanAnnoation(annotations: Annotations, annotation: string): number {
+    return annotations[annotation] && annotations[annotation] === 'no' ? 0 : 1
```

Comment:
I tried this with a node that only had the caasp version annotation and it came up with 1 node needing upgrade with a disruptive and a security update. In cases where it's missing should this return 0?